wheatley/bot.py

```diff
@@ -112,13 +112,30 @@ def _on_row_gen_change(self, row_gen_json):
             self.logger.warning(e)
 
     def _on_size_change(self):
-        if not self.row_generator.is_tower_size_valid(self._tower.number_of_bells):
-            self.logger.warning(f"Row generation requires {self.row_generator.number_of_bells} \
-bells, but the current tower has {self._tower.number_of_bells}.  Wheatley will crash when you go \
-into changes unless something is done!")
+        self._check_number_of_bells()
+
+    def _check_number_of_bells(self):
+        """ Returns whether Wheatley can ring with the current number of bells with reasons why not"""
```

Comment:
Nitpick: This'd look cleaner and more balanced if it had a space before the closing `"""`.

Reply:
```suggestion
        """ Returns whether Wheatley can ring with the current number of bells with reasons why not """
```
You can add a suggestion (Ctrl+G or just ` ```suggestion` and type)

Reply:
Aha that's how you do it!  I couldn't figure it out - it was just black magic as far as I was concerned!